MapboxSearch/base/src/main/java/com/mapbox/search/base/result/SearchResultFactory.kt

```diff
@@ -85,8 +86,25 @@ class SearchResultFactory(private val recordResolver: IndexableRecordResolver) {
                 }
             }
             CoreApiType.SEARCH_BOX -> {
-                callback(Result.failure(Exception("Search Box is not supported yet")))
-                return AsyncOperationTaskImpl.COMPLETED
+                if (searchResult.type == BaseRawResultType.BRAND) {
+                    val result = if (searchResult.isValidBrandType) {
+                        val value = BaseServerSearchSuggestion(searchResult, requestOptions)
+                        Result.success(value)
+                    } else {
+                        Result.failure(InternalIgnorableException("Skipping invalid BRAND search result"))
+                    }
+                    callback(result)
+                    return AsyncOperationTaskImpl.COMPLETED
+                } else if (searchResult.action == null) {
+                    if (searchResult.type == BaseRawResultType.QUERY) {
+                        callback(Result.failure(InternalIgnorableException("Skipping query suggestion without action")))
+                        return AsyncOperationTaskImpl.COMPLETED
+                    } else if (searchResult.type != BaseRawResultType.USER_RECORD) {
+                        failDebug { "Can't create search suggestion from. ${debugInfo()}" }
+                        callback(Result.failure(Exception("Can't create search suggestion from $searchResult")))
+                        return AsyncOperationTaskImpl.COMPLETED
+                    }
+                }
```

Comment:
When is an invalid BRAND returned? How common is this? What does this part mean `else if (searchResult.action == null)`? When is there not an action?